src/use_case/vote_out/VoteOutInteractor.java

```diff
@@ -1,4 +1,60 @@
 package use_case.vote_out;
 
+import entity.Game;
+import entity.PromptGenerator;
+import use_case.ChatAPIAccessInterface;
+
 public class VoteOutInteractor implements VoteOutInputBoundary {
+    private final VoteOutDataAccessInterface gameDataAccessObject;
+    private final ChatAPIAccessInterface storyDataAccessObject;
+    private final VoteOutOutputBoundary userPresenter;
+    private final Game game;
+    private final PromptGenerator promptGenerator;
+
+
+    public VoteOutInteractor(VoteOutDataAccessInterface gameDataAccessObject, ChatAPIAccessInterface storyDataAccessObject, VoteOutOutputBoundary userPresenter) {
+        this.gameDataAccessObject = gameDataAccessObject;
+        this.storyDataAccessObject = storyDataAccessObject;
+        this.userPresenter = userPresenter;
+        // Get the game
+        game = gameDataAccessObject.getGame();
+        // Get the prompt generator
+        promptGenerator = gameDataAccessObject.getPromptGenerator();
+    }
+
+    // This method prepares success view if a werewolf or villager is voted out
+    // Meanwhile, if an incorrect username is entered (it doesn't exist) the fail view is prepared
+    @Override
+    public void voteOutPlayer(VoteOutInputData voteOutInputData) {
+        // Get name of player voted out from input data
+        String playerVotedOut = voteOutInputData.getVotedName();
+        // This case is specifically when the name isn't a werewolf nor is it a player, so the name doesn't exist
+        if (!(game.getAliveVillagers().containsKey(playerVotedOut) || game.getAliveWerewolves().containsKey(playerVotedOut))) {
+            // TODO: we can make "no such player exists" a constant or something later but not important right now
+            userPresenter.prepareFailView("No such player exists");
```

Comment:
For prepareFailView in KillVillager, I left the message up to the Presenter to implement as I thought it would make more sense for us to edit the actual message there instead of in the use case interactor. Let me know what you think.

Reply:
This is just a preference thing. I feel like it makes way more sense here because the interactor is what actually goes through the logic of what the error is so it should be the one to name the error, and again that's what they did in week5ca.